python/pyarrow/array.pyx

```diff
@@ -217,11 +220,13 @@ def from_pylist(object list_obj, DataType type=None):
     return box_arrow_array(sp_array)
 
 
-def from_pandas_series(object series, object mask=None):
+def from_pandas_series(object series, object mask=None, timestamps_to_ms=False):
```

Comment:
I still see this argument as "temporary". Can you create a JIRA to store native int64 nanoseconds? We should also add an argument for more generalized type casting options (to avoid an excess of one-off arguments)